io/src/main/scala/laika/io/model/Input.scala

```diff
@@ -78,6 +82,12 @@ object TextInput {
     TextInput[F](path, docType, InputRuntime.textFileResource[F](file, codec).map(StreamReader(_, file.length.toInt)), Some(file))
   def fromStream[F[_]: Sync] (path: Path, docType: TextDocumentType, stream: F[InputStream], codec: Codec, autoClose: Boolean): TextInput[F] =
```

Comment:
Btw, I think all uses of `F[InputStream]` should actually be replaced with `fs2.Stream[F, Byte]`. It is a referentially transparent, pure-FP, and `fs2.io` includes methods to get `fs2.Stream` from classpath resources etc. while correctly handling blocking.

Reply:
Yes, I feel the same. However, I think we should be careful because  i/o stream is  such a fundamental component for io module that it will propagate changes.

Reply:
Unfortunately I have not yet looked at this PR in detail, but as a general tendency, I feel like while it would make sense to standardise on `fs2.Stream` for any *internal* API wherever feasible, I think I'd still like to offer `InputStream` in the public APIs, since it is such a common type and this lib is also intended for an audience that does not know that much about Typelevel libs. The implementation of such a method could then immediately translate down to fs2 wherever possible.

Reply:
> I think I'd still like to offer `InputStream` in the public APIs, since it is such a common type and this lib is also intended for an audience that does not know that much about Typelevel libs.

:100: I definitely agree with this sentiment, with regard to accessibility.

However, in this case I would argue that signatures `F[InputStream]` are actually more likely to be abused by the uninformed user than `fs2.Stream[F, Byte]`.

A user who doesn't know much about Cats Effect could very plausibly write `F.pure` where they meant `F.delay` or even worse `F.blocking` and cause all sorts of problems for Laika/Cats Effect which make many important assumptions about the semantics of code suspended in `F[_]` (for example, at which point side-effects vs `pure` are evaluated and on which threadpool).

On the other hand, It seems reasonable to add scaladocs to point a user to methods such as `fs2.io.readClassResource("myResource")` to get an `fs2.Stream[F, Byte]`, which I think are easy to use, hard to abuse, and do the right thing out-of-the-box.

To offer an accessible "unsafe" API, FWIW I think you should do away with `F[_]` entirely which I suspect is confusing for this group of users as well. This API can be then implemented in an imperative style and take the appropriate measures not to violate the assumptions of the Cats Effect runtime.

Reply:
I was actually thinking about the specific use case of interop with existing tools that provide an `InputStream`, not the simpler scenarios of classpath or file resources. Removing the existing `fromStream` method does not make anything safer here, since it just moves the problem to `fs2.io.readInputStream` which also expects an `F[InputStream]`. The way I see it it's just more code to write for the user and the same kind of trap to fall into.

My preference would be not to remove any existing API, but instead only add `fromStream(fs2.io.Stream)` and then use fs2 for most of the internal API. Which means to also keep `fromFile(String)` and `fromClasspathResource(String)`.

Reply:
But it's a good point to improve scaladoc for users not familiar with cats-effect. From working with junior devs in my day job I am aware that confusing `F.delay`, `F.pure` and `F.blocking` is amongst the no1 sources of bugs when using effects (to some degree even for seniors, to my surprise).